R/init.R

```diff
@@ -302,9 +302,23 @@ renv_init_repos <- function() {
     return(repos)
   }
 
-  # if no repository was set, use PPM
-  if (identical(repos, list(CRAN = "@CRAN@")))
-    return(config$ppm.url())
+  # otherwise, check for some common 'default' CRAN settings
+  cran <- repos[["CRAN"]]
+  if (is.character(cran) && length(cran) == 1L) {
+    cran <- sub("/*$", "", cran)
+    defaults <- c(
+      "@CRAN@",
+      "https://cloud.R-project.org",
+      "https://cran.rstudio.com",
+      "https://cran.rstudio.org"
+    )
+
+    if (tolower(cran) %in% tolower(defaults)) {
+      repos[["CRAN"]] <- config$ppm.url()
+      return(repos)
+    }
```

Comment:
```suggestion
  std <- sub("/*$", "", repos)
  defaults <- c(
    "@CRAN@",
    "https://cloud.R-project.org",
    "https://cran.rstudio.com",
    "https://cran.rstudio.org"
  )

  repos[tolower(cran) %in% tolower(std)] <- config$ppm.url()
```

Not sure if this is any better, but maybe it's a bit easier to reason about?